src/qibo/abstractions/gates.py

```diff
@@ -46,6 +46,7 @@ def __init__(self, q):
 
     def controlled_by(self, *q):
         """Fall back to CNOT and Toffoli if there is one or two controls."""
+        self._controlled_by_error()
```

Comment:
Optiona: shall we convert this to a decorator?

Reply:
I changed this to a decorator. This was actually a good example to play with decorators, thanks!